sdk/python/lib/pulumi/runtime/_json.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2016-2020, Pulumi Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import abc
+import json
+from typing import Any
+
+from .. import _types
+
+
+def _to_json_serializable_obj(obj: Any) -> Any:
```

Comment:
I'm curious if we considered using pickle here, I believe it's capable of serializing class instances. https://docs.python.org/3/library/pickle.html#pickling-class-instances